sql/sql_expression.cc

```diff
@@ -1539,16 +1539,21 @@ bool
 SqlExpression::
 isConstant() const
 {
-    return false;
+    for (auto & c: getChildren()) {
+        if (!c->isConstant())
+            return false;
+    }
+    return true;
 }
 
 ExpressionValue
 SqlExpression::
 constantValue() const
 {
-    throw HttpReturnException(400, "Expression is not provably constant",
-                              "surface", surface,
-                              "ast", print());
+    SqlExpressionConstantScope scope;
```

Comment:
The isConstant has a slightly tighter scope (eg, it's possible that something is a constant because it has a case which avoids the evaluation of  non-constant part).  I will improve the error message.